File: backend/app/schemas/flashcard.py
from __future__ import annotations

from enum import Enum
from typing import Optional

from pydantic import BaseModel, Field


class FlashcardMode(str, Enum):
    all = "all"
    review_status = "review_status"
    category = "category"


class FlashcardCard(BaseModel):
    user_word_id: int
    word: str
    definition: str
    phonetic: Optional[str] = None
    audio_url: Optional[str] = None


class FlashcardsResponse(BaseModel):
    total: int
    cards: list[FlashcardCard]


class FlashcardsCompleteRequest(BaseModel):
    user_word_ids: list[int] = Field(default_factory=list)

File: frontend/src/types/flashcard.ts
export type FlashcardMode = "all" | "review_status" | "category";
export type ReviewStatus = "NEWBIE" | "LEARNING" | "MASTERED";


export interface FlashcardsResponse {
  total: number;
  cards: FlashcardCard[];
}

export interface FlashcardCard {
  user_word_id: number;
  word: string;
  definition: string;
  phonetic?: string | null;
  audio_url?: string | null;
}

export interface FlashcardsCompleteRequest {
  user_word_ids: number[];
}
